src/common/args.h

```diff
@@ -291,7 +425,6 @@ class ArgsManager
      * @return command-line argument (0 if invalid number) or default value
      */
     int64_t GetIntArg(const std::string& strArg, int64_t nDefault) const;
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/16545#discussion_r1718228472

> The example for `ALLOW_INT | ALLOW_BOOL` in 6865a198f5db30bd494b3a2540f47ee728963908 is a bit worrying.

Thanks, and yes that example is very out of date. I removed it from the PR description. Now that std::optional GetArg overloads exist, there is little reason to use the other GetArg functions, and their behavior has also been simplified. The way this would be written with the current PR would look more like the `-rescan` example from example_tests, and should be more readable:

https://github.com/bitcoin/bitcoin/blob/1e37bcf9fc11562baaedea24685c31f60ef2de31/src/test/argsman_tests.cpp#L145-L153